### PR TITLE
Now the script will store credentials upon successful OAuth

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Now the script will store credentials upon successful OAuth, through which we can use the same credentials even after the server restarts.

The system has been changed now. The token system now relies on a database to store it and is wrapped in routes for easy operation.

### CODE REVIEW

**Comment by awasthishubh** [comment]
Many hosting platform (git based like heroku) don't allow changes to local files. Use database.

**Comment by YashKumarVerma** [comment]
@awasthishubh ,

The system has been changed now. The token system now relies on a database to store it and is wrapped in routes for easy operation.

**Comment by awasthishubh** [comment]
This process still requires user to go to step1 route.
Optimal solution will be
* Start the server
* Load tokens from db
* If tokens don't exist, open the oauth route in browser (or give user the link in terminal only).
This whole process should start as soon as server starts. Since it's mandatory to do this don't rely on user to open any route.

File: backend/bin/database/connect.js [hunk 1]
```diff
@@ -3,25 +3,31 @@ require('dotenv').config();
...
-        console.log('Database connection successful');
-      })
-      .catch(err => {
-        console.error('Database connection error');
-      });
-  }
+	_connect() {
+		mongoose
+			.connect(
+				`mongodb://${process.env.DB_HOST}/${process.env.DB_NAME}`,
```
Comment (awasthishubh) [inline backend/bin/database/connect.js]: What if db authentication requires username/password?

**Comment by YashKumarVerma** [comment]
> This process still requires user to go to step1 route.
> Optimal solution will be
> 
> * Start the server
> * Load tokens from db
> * If tokens don't exist, open the oauth route in browser (or give user the link in terminal only).
> 
> This whole process should start as soon as server starts. Since it's mandatory to do this don't rely on user to open any route.

This issue has been fixed in the version I'm working on. The Spotify object has a method named seed, which automatically loads the tokens onto the object wherever they're being used.

File: backend/bin/database/connect.js (hunk 1 above)
Comment (YashKumarVerma) [reply backend/bin/database/connect.js]: Will work on it.

File: backend/bin/database/connect.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ require('dotenv').config();
 const mongoose = require('mongoose');
 const logger = require('./../logger/logger');
 
+logger.info(`mongodb://${process.env.DB_USER}:${process.env.DB_PASS}@${process.env.DB_HOST}/${process.env.DB_NAME}`);
```
Comment (YashKumarVerma) [inline backend/bin/database/connect.js]: would be removed in next PR

**Pull request merged**